lib/openstack_api/openstack_server.py

```diff
@@ -0,0 +1,286 @@
+from typing import List
+
+from openstack.compute.v2.server import Server
+
+from openstack_api.openstack_connection import OpenstackConnection
+from openstack_api.openstack_identity import OpenstackIdentity
+from openstack_api.openstack_query import OpenstackQuery
+from openstack_api.openstack_wrapper_base import OpenstackWrapperBase
+
+
+class OpenstackServer(OpenstackWrapperBase):
+    # Lists all possible query presets for server.list
+    SEARCH_QUERY_PRESETS: List[str] = [
+        "all_servers",
+        "servers_older_than",
+        "servers_younger_than",
+        "servers_last_updated_before",
+        "servers_last_updated_after",
+        "servers_id_in",
+        "servers_id_not_in",
+        "servers_name_in",
+        "servers_name_not_in",
+        "servers_name_contains",
+        "servers_name_not_contains",
+        "servers_errored",
+        "servers_shutoff",
+        "servers_errored_and_shutoff",
+    ]
+
+    # Lists possible queries presets that don't require a project to function
+    SEARCH_QUERY_PRESETS_NO_PROJECT: List[str] = [
+        "servers_older_than",
+        "servers_last_updated_before",
+    ]
+
+    def __init__(self, connection_cls=OpenstackConnection):
+        super().__init__(connection_cls)
+        self._identity_api = OpenstackIdentity(self._connection_cls)
+        self._query_api = OpenstackQuery(self._connection_cls)
+
+    def __getitem__(self, item):
+        return getattr(self, item)
+
+    def search_all_servers(
+        self, cloud_account: str, project_identifier: str, **_
+    ) -> List[Server]:
+        """
+        Returns a list of Servers matching a given query
+        :param cloud_account: The associated clouds.yaml account
+        :param project_identifier: The project to get all associated servers with, can be empty for all projects
+        :return: A list of all servers
+        """
+        selected_servers = []
+        if project_identifier == "":
+            projects = self._identity_api.list_projects(cloud_account)
+        else:
+            projects = [
```

Comment:
I think you can only search a project at a time looking at the params, any reason why we pass the projects around as a list?

Reply:
Yes as it will iterate over all of them when searching for servers.